airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -302,6 +312,18 @@ def on_access_denied_job(self, **kwagrs) -> AirbyteTracedException:
     def on_job_with_errors(self, errors: List[Mapping[str, Any]]) -> AirbyteTracedException:
         raise ShopifyBulkExceptions.BulkJobUnknownError(f"Could not validate the status of the BULK Job `{self.job_id}`. Errors: {errors}.")
 
+    def on_retryable_error(self, response: requests.Response, exception: Exception) -> Optional[requests.Response]:
+        if self._job_retry_on_error_count == self._job_retry_on_error_max_limit:
```

Comment:
So my understanding is that this job retry mechanism is shared for all the slices. So if we have 10 slices and each fails only once, the stream will be incomplete, is that correct? I'm wondering if we should change that or update the naming of the variable so that it's clear that it is shared for all the slices

Reply:
We have the `stream_slices()` as a generator that works with 1 slice/time. If the slice faces the retryable Error - we go with the retry until the `_job_retry_on_error_max_limit ` is reached; otherwise, we raise the Error.

Reply:
That makes sense to me but don't we share the same job_manager for all the slices in a stream? It seems like all slices should be able to fail at least once without the stream breaking but I don't have much to confirm that. Example:

```
* Processing slice 1:
    * First call fail
    * Retry succeed
* Processing slice 2:
    * First call fail
    * Retry succeed
...
* Processing slice n:
    * First call fail
    * Retry succeed
```

In the case, since the API is still working from time to time, it seems like we should continue. I agree though that we should avoir cases like:
```
* Processing slice 1:
    * First call fail
    * Retry fail
    * Retry fail again
    <...>
```

Reply:
That is correct, if the retried slice is succeeded - we go for the next one, parsing the result for the succeeded one before that.

Reply:
But `_job_retry_on_error_count` is never reset to 0 between the slices, right? This would mean that the slices share the same count

Reply:
Oh, got it, finally! 
Added the reset for that counter when we successfully got out of the loop with the successful request after retry.